File: typescript/tests/integration/mobile-get-all-ui-elements-xml.integration.test.ts
/**
 * Integration tests for Mobile.getAllUIElements XML format.
 *
 * Note: This test creates a real session. Do not run concurrently.
 */

import { AgentBay } from '../../src/agent-bay';
import { log } from "../../src/utils/logger";

describe('Mobile GetAllUIElements XML Integration Tests', () => {
  let agentBay: AgentBay;
  const apiKey = process.env.AGENTBAY_API_KEY;

  beforeAll(() => {
    if (!apiKey) {
      console.warn('Warning: AGENTBAY_API_KEY not set. Tests will be skipped.');
    }
  });

  beforeEach(() => {
    if (apiKey) {
      agentBay = new AgentBay({ apiKey });
    }
  });

  test('should return XML raw output for get_all_ui_elements', async () => {
    if (!apiKey) {
      log('Skipping test: AGENTBAY_API_KEY not set');
      return;
    }

    const sessionResult = await agentBay.create({
      imageId: 'imgc-0ab5takhnlaixj11v'
    });

    expect(sessionResult.session).toBeDefined();
    const session = sessionResult.session!;

    try {
      await new Promise(resolve => setTimeout(resolve, 15000));

      const ui = await session.mobile.getAllUIElements(10000, 'xml');

      expect(ui.success).toBe(true);
      expect(ui.requestId).toBeDefined();
      expect(ui.requestId.length).toBeGreaterThan(0);
      expect(ui.format).toBe('xml');
      expect(typeof ui.raw).toBe('string');
      expect(ui.raw.trim()).not.toBe('');
      expect(ui.raw.trim().startsWith('<?xml')).toBe(true);
      expect(ui.raw).toContain('<hierarchy');
      expect(ui.elements).toEqual([]);
    } finally {
      await session.delete();
    }
  }, 60000);
});

File: golang/tests/pkg/integration/mobile_get_all_ui_elements_xml_integration_test.go
package integration

import (
	"os"
	"testing"
	"time"

	"github.com/aliyun/wuying-agentbay-sdk/golang/pkg/agentbay"
	"github.com/stretchr/testify/assert"
	"github.com/stretchr/testify/require"
)

func TestMobileGetAllUIElementsXMLFormatContract(t *testing.T) {
	apiKey := os.Getenv("AGENTBAY_API_KEY")
	if apiKey == "" {
		t.Skip("Skipping integration test: AGENTBAY_API_KEY not set")
	}

	agentBay, err := agentbay.NewAgentBay(apiKey)
	require.NoError(t, err, "Failed to create AgentBay client")

	sessionParams := &agentbay.CreateSessionParams{
		ImageId: "imgc-0ab5takhnlaixj11v",
	}
	sessionResult, err := agentBay.Create(sessionParams)
	require.NoError(t, err, "Failed to create session")
	require.NotNil(t, sessionResult.Session, "Session should be created")

	session := sessionResult.Session
	defer func() {
		_, _ = session.Delete()
	}()

	time.Sleep(15 * time.Second)

	ui := session.Mobile.GetAllUIElements(10000, "xml")
	assert.NotEmpty(t, ui.RequestID, "Should have request ID")
	assert.Empty(t, ui.ErrorMessage, "get_all_ui_elements(xml) should not return error")
	assert.Equal(t, "xml", ui.Format)
	assert.NotEmpty(t, ui.Raw)
	assert.Contains(t, ui.Raw, "<hierarchy")
	assert.Equal(t, 0, len(ui.Elements))
}
